File: android/app/src/main/java/com/naivesound/scherzo/MidiKeyView.java
package com.naivesound.scherzo;

import android.content.Context;
import android.graphics.Rect;
import android.util.Log;
import android.view.MotionEvent;
import android.view.View;

public class MidiKeyView extends View {

    private final static String TAG = "MidiKeyView";

    public MidiKeyView(Context context) {
        super(context);
    }

    @Override
    public boolean onTouchEvent(MotionEvent event) {
        int note = (Integer) getTag();
        if (event.getAction() == MotionEvent.ACTION_DOWN) {
            App.getService().noteOn(0, note, 127);
        }
        if (event.getAction() == MotionEvent.ACTION_UP ||
                event.getAction() == MotionEvent.ACTION_CANCEL) {
            App.getService().noteOff(0, note);
        }
        return true;
    }
}

File: android/app/src/main/java/com/naivesound/scherzo/MidiKeyView.kt
package com.naivesound.scherzo

import android.content.Context
import android.view.MotionEvent
import android.view.View

class MidiKeyView(context: Context) : View(context) {

    override fun onTouchEvent(event: MotionEvent): Boolean {
        val note = tag as Int
        if (event.action == MotionEvent.ACTION_DOWN) {
            App.getService()?.noteOn(0, note, 127)
        }
        if (event.action == MotionEvent.ACTION_UP || event.action == MotionEvent.ACTION_CANCEL) {
            App.getService()?.noteOff(0, note)
        }
        return true
    }
}
